app/views/admin/attachments/_form.html.erb

```diff
@@ -1,12 +1,34 @@
 <% initialise_script "GOVUK.AdminAttachmentsForm", selector: '.js-attachment-form', right_to_left_locales: Locale.right_to_left.collect(&:to_param) %>
 
 <%= form_for attachment, url: [:admin, typecast_for_attachable_routing(attachable), attachment.becomes(Attachment)], as: :attachment, html: { :class => "js-attachment-form" } do |form| %>
-  <%= form.errors %>
-
-  <%= content_tag :fieldset, class: ("right-to-left" if form.object.rtl_locale?) do %>
-    <%= form.text_field :title %>
+  <% if form.object.errors.any? %>
+    <%= render "govuk_publishing_components/components/error_summary", {
+      title: "To save the #{attachment.readable_type} attachment please fix the following issues:",
+      items: form.object.errors.full_messages.map do |msg|
+        {
+          text: msg
+        }
+      end
+    } %>
   <% end %>
 
+  <div class="govuk-!-margin-bottom-8">
+    <%= render "govuk_publishing_components/components/input", {
+      label: {
+        text: "Title (required)"
+      },
+      name: "attachment[title]",
+      heading_level: 2,
+      heading_size: "l",
+      value: form.object.title,
+      error_items: (form.object.errors.as_json(full_messages: true)[:title].map do |error|
```

Comment:
I've added a helper method here for this https://github.com/alphagov/whitehall/blob/ed6c5d2a4a4438387f4d8e4cbfc21ffe937e33cf/app/helpers/errors_helper.rb

You can call it by doing the following

```
errors_for(object.errors, :attribute)
```

So in this case it would be 

```
errors_for(attachment.errors, :title)
```

Reply:
Yours does something v similar though 👍

Reply:
Updated to use the helper.